xocto/types/generic.py

```diff
@@ -0,0 +1,21 @@
+from typing import Any, Protocol, TypeVar
+
+
+T = TypeVar("T", covariant=True)
+
+
+class Comparable(Protocol[T]):
+    """
+    Just a very basic way of describing an object that can be compared to another using some
```

Comment:
```suggestion
    A very basic way of describing an object that can be compared to another using some
```

irrelevant to your change.... but....

Reply:
Good call out - I considered making that change at the same time too ;)

Reply:
You could get rid of `very` for a more concise sentence with the same meaning to readers. Very works better to the ear than it does to the eye.

Reply:
I shortened it to "A way of.." which I think reads better 👍